### ISSUE TITLE
Move code to Scala 2.13 and eventually 3 (it's on 2.12 currently)

### ISSUE THREAD

**Issue by SethTisue** [VirtusLab/scala-cli#739]
At present scala-cli is built with Scala 2.12:

```
[info] welcome to sbt 1.6.2 (Eclipse Adoptium Java 17.0.2)
sbt:scala-cli> show scalaVersion
[info] 2.12.15
```

@romanowski says the intention is to upgrade to 2.13 relatively soon, and eventually to 3.

If I understood correctly, he said that there aren't any major roadblocks to getting on 2.13, but getting to 3 would be harder because the codebase uses scala-async, which is 2-only at present.

In any case, I thought there should be a ticket documenting the present situation and future intentions.

**Comment by lwronski** [comment]
Hi @SethTisue, it seems that `scala-cli` uses 2.13 to build itself - [default internal version](https://github.com/VirtusLab/scala-cli/blob/93a13c862e464433e077d6dbd690f6180708e3ea/project/deps.sc#L28).

But the latest version of Scala CLI `0.1.2` was released with Scala `2.12`, so the `0.1.3` should be released with `2.13`.

**Comment by SethTisue** [comment]
Oh, good news!

I think I'll close this then, since it was mainly the 2.13 part that concerned me.
